specification/automation/resource-manager/Microsoft.Automation/stable/2015-10-31/definitions.json

```diff
@@ -535,6 +536,11 @@
           "readOnly": true,
           "description": "Gets the name of the certificate."
         },
+        "type": {
+          "type": "string",
+          "readOnly": true,
+          "description": "Resource type"
+        },
```

Comment:
The proper way to fix this is, to split the `Resource` definition into 3

```json
    "Resource": {
      "description": "The core properties of ARM resources",
      "properties": {
        "id": {
          "readOnly": true,
          "type": "string",
          "description": "Fully qualified resource Id for the resource."
        },
        "name": {
          "readOnly": true,
          "type": "string",
          "description": "The name of the resource"
        },
        "type": {
          "readOnly": true,
          "type": "string",
          "description": "The type of the resource."
        }
      },
      "x-ms-azure-resource": true
    },
    "TrackedResource": {
      "description": "The resource model definition for a ARM tracked top level resource",
      "properties": {
        "tags": {
          "type": "object",
          "additionalProperties": {
            "type": "string"
          },
          "x-ms-mutability": [
            "read",
            "create",
            "update"
          ],
          "description": "Resource tags."
        },
        "location": {
          "type": "string",
          "x-ms-mutability": [
            "read",
            "create"
          ],
          "description": "The Azure Region where the resource lives"
        }
      },
      "allOf": [
        {
          "$ref": "#/definitions/Resource"
        }
      ]
    },
    "ProxyResource": {
      "description": "The resource model definition for a ARM proxy resource. It will have everything other than required location and tags",
      "allOf": [
        {
          "$ref": "#/definitions/Resource"
        }
      ]
    }
```

And all top level resource to derive from the  `TrackedResource` and nested resource (like Certificates) to derive from `ProxyResource`, we shouldn't annotate arbitrary types with `x-ms-azure-resource`.

Reply:
@johanste can you chime in and confirm whether it make sense to follow the above pattern?

Reply:
thanks @johanste for the response and correcting the confusion around nested resource.

Since this is an existing API version, conforming to this inheritance hierarchy will cause breaking change (for the models which are based on older Resource base type). 

@DeveloperTommy can you please create a work item at your end so that in the next api version we can ensure the models uses this standard hierarchy? Also make a note on the current RPCViolations and SDKViolations [reported](https://travis-ci.org/Azure/azure-rest-api-specs/jobs/335266447) by CI, many of them will go away once you conform to this hierarchy.

Reply:
From the reviewer side I will create an issue in the spec repo and ref this PR.

Reply:
@anuchandy Yes I have just created the work item Task 139186: Refactor Swagger model to follow standard hierarchy and noted the violations from Travis.

Reply:
@anuchandy @johanste Bumping for my previous comment.

Reply:
@anuchandy, do you have anything else you want to get addressed?

Reply:
no, looks good.. merging this.

Reply:
Thank you both for reviewing!